Obsidian/Net/MinecraftStream.cs

```diff
@@ -255,7 +255,14 @@ public async Task WriteAsync(DataType type, FieldAttribute attribute, object val
                     }
                 case DataType.Float:
                     {
-                        await this.WriteFloatAsync((float)value);
+                        if (value is Enum)
+                        {
+                            Enum _enum = (Enum)value;
+                            value = Convert.ChangeType(value, _enum.GetTypeCode());
+                            await this.WriteFloatAsync(Convert.ToSingle(value));
+                        }
+                        else
+                            await this.WriteFloatAsync((float)value);
```

Comment:
I don't see why `float`'s should be ever mapped to `enum`s, is this the case anywhere? This doesn't have to removed though.

Reply:
ChangeGameState uses floats for its values ^ ^

Reply:
I think this should be based on what type the enum uses under the hood (`enum thing : int`, or `enum thing : float`, etc)

Reply:
It can be an int. I checked to see if I could make it a float enum, but c# doesn't support it.

Reply:
If that's the case, leave it in.

![blobcatcomfglare](https://user-images.githubusercontent.com/22963120/99849738-62e44c00-2b7c-11eb-82de-09d6e7bad24a.png)

Reply:
ight, lgtm then